ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ConnectExecutor.java

```diff
@@ -81,6 +92,44 @@ public static StatementExecutorResponse execute(
         .map(err -> new ErrorEntity(statement.getStatementText(), err)));
   }
 
+  private static List<String> validate(final CreateConnector createConnector,
+      final ConnectClient client) {
+    final Map<String, String> config = new HashMap<>(createConnector.getConfig().size());
+    createConnector.getConfig().forEach((k, v) -> config.put(k, v.toString()));
+    final String connectorType = config.get("connector.class");
+    // Request with an empty connector type in the url results in 404.
+    // Request with a connector type that contains only spaces results in 405.
+    // In both cases it is user-friendlier to return the actual error.
+    if (connectorType == null) {
+      // Return error message identical to the one produced by the connector
+      return ImmutableList.of(String.format(
+          "Connector config %s contains no connector type", config));
+    } else if (connectorType
+        .replace("'", "")
+        .replace("\"", "")
```

Comment:
Out of curiosity, why replace quotes in particular (and not other special characters as well)?

Reply:
Consider `shouldReturnErrorIfConnectorTypeIsEmpty`:

https://github.com/confluentinc/ksql/blob/d2db98c9ac464d60a82c1475e7a23c64c03c5528/ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/ConnectExecutorTest.java#L215-L235

In this case `connectorType` will look something like `' '` - a whitespace surrounded by single quotes. 

I didn't remove other special characters because the purpose of these transformations is to hedge against sending an empty connector type(i.e. `""` or a string containing only whitespaces). As for different unicode space characters, Java's `String#trim` is supposed to remove them. For a peace of mind, I tested a couple of those and it worked fine :).